client/components/domains/domain-search-results/index.jsx

```diff
@@ -290,6 +293,7 @@ class DomainSearchResults extends React.Component {
 						unavailableDomains={ this.props.unavailableDomains }
 						isEligibleVariantForDomainTest={ this.props.isEligibleVariantForDomainTest }
 						shouldHideFreeDomainExplainer={ this.props.shouldHideFreeDomainExplainer }
+						buttonStyles={ { primary: isReskinnedAndNotOnMobile } }
```

Comment:
> It could be related to @klimeryk's #44223 (comment). (Thanks @klimeryk for the review!)
> We show the reskinned UI whenever a user is assigned to the variant. From this discussion I understand that signup persists some state and there might be other checks that would need to be applied.

The signup state is cleared when the signup completes(though there are some edge cases where it can persist but that could be tangential to this discussion). I think this issue is more because the domain component that is being modified by this PR is shared by different views - signup view, launch view, domain management view,among others. So, the line I've referenced above which sets button style as primary for the reskinned flow looks suspect because this would set that style for all the views that share this component. 

I think you might have to add an additional check like `isSignupStep` whose value is present in that component.

Reply:
Yes, that makes sense and I was able to reproduce this issue by viewing the domains page as a logged in user.

I have added this check in https://github.com/Automattic/wp-calypso/pull/44223/commits/3ef43b2bdf6e27f176ab0b1ccea227d9c95c2795.

However, if the domains step is shown in any other signup flow, and the user happens to be assigned to this variation, we still end up showing the new UI.

@niranjan-uma-shankar Can you please help me with accessing & testing signup flows other than the default (`onboarding` I presume)?

Reply:
I think the check for `onboarding` flow name [in this line](https://github.com/Automattic/wp-calypso/blob/3ef43b2bdf6e27f176ab0b1ccea227d9c95c2795/client/signup/main.jsx#L696) takes care of this. You can try going to /start/premium logged in, or the site launch flow and check if the reskinned UI is showing up. 

The signup flows are listed [here](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/flows-pure.js), but the relevant flows to us would be /free, /personal, /premium, /businesss, /ecommerce, /domain (this is the domain only flow),